### PR TITLE
Bump bootstrap to 1.60

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
This bumps the bootstrap compiler to 1.60 and cleans up cfgs and Span's rustc_pass_by_value (enabled by the bootstrap bump).

### CODE REVIEW

File: library/core/src/panicking.rs [hunk 1]
```diff
@@ -85,9 +84,9 @@ fn panic_bounds_check(index: usize, len: usize) -> ! {
     panic!("index out of bounds: the len is {} but the index is {}", len, index)
 }
 
-#[cfg(not(bootstrap))]
 #[cold]
 #[inline(never)]
+#[cfg_attr(bootstrap, track_caller)]
```
Comment (bjorn3) [inline library/core/src/panicking.rs:89]: Why is this necessary? Why only for the bootstrap compiler?

**Comment by pietroalbini** [comment]
r=me

File: library/core/src/panicking.rs (hunk 1 above)
Comment (Mark-Simulacrum) [reply library/core/src/panicking.rs:89]: See the CI failures - something in the bootstrap compiler is expecting to pass caller information to this function, but that has been dropped in non-bootstrap compilers.

I haven't deeply investigated but I suspect it's related to recent work here

Comment (bjorn3) [reply library/core/src/panicking.rs:89]: Why should non-bootstrap compilers not use it? Not using `#[track_caller]` makes the `Location::caller()` below point to this function, which is not very useful.

**Comment by mati865** [comment]
CI error:
```
warning: spurious network error (1 tries remaining): failed to get 200 response from `[https://crates.io/api/v1/crates/libc/0.2.116/download`,](https://crates.io/api/v1/crates/libc/0.2.116/download%60,) got 502
```

**Comment by c410-f3r** [comment]
`crates.io` is having some problems

File: library/core/src/panicking.rs (hunk 1 above)
Comment (eddyb) [reply library/core/src/panicking.rs:89]: Sounds like that feature is somewhat broken then, and wasn't properly tested. In fact, it looks like it never had `#[track_caller]` before? Personally I would probably open an issue (pinging everyone involved) and leave a FIXME comment here - while this is bad, it doesn't really make sense to block a bootstrap bump on it.

Comment (Mark-Simulacrum) [reply library/core/src/panicking.rs:91]: I filed https://github.com/rust-lang/rust/issues/94337 and have pushed a commit adding a FIXME.

File: library/core/src/panicking.rs [hunk 2]
```diff
@@ -84,8 +84,12 @@ fn panic_bounds_check(index: usize, len: usize) -> ! {
     panic!("index out of bounds: the len is {} but the index is {}", len, index)
 }
 
+// This function is called directly by the codegen backend, and must not have
+// any extra arguments (including those synthesized by track_caller). It should
+// also not be called by regular Rust code.
```
Comment (nbdd0121) [inline library/core/src/panicking.rs:89]: It's okay to call this in regular Rust code, it's just that it doesn't really make much sense.
